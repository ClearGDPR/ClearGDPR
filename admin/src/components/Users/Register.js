import React from 'react';
import PropTypes from 'prop-types';
import { Form } from 'informed';
import TextInput from 'components/core/Common/Forms/TextInput';
import Loader from 'components/core/cards/dashboard/Loader';

export class Register extends React.Component {
  static propTypes = {
    isLoading: PropTypes.bool,
    onSubmit: PropTypes.func,
    validatePassword: PropTypes.func,
    touched: PropTypes.object,
    errors: PropTypes.object
  };

  renderForm() {
    return (
      <React.Fragment>
        <TextInput
          label="Username"
          placeholder="mick.bern"
          error={
            this.props.touched &&
            this.props.touched['username'] &&
            this.props.errors &&
            this.props.errors['username']
          }
          field="username"
        />
        <TextInput
          label="New password"
          placeholder="*********"
          type={'password'}
          error={
            this.props.touched &&
            this.props.touched['password'] &&
            this.props.errors &&
            this.props.errors['password']
          }
          field="password"
          validate={this.props.validatePassword}
        />
        <div>
          <input type="submit" className="btn" value="Register" />
        </div>
      </React.Fragment>
    );
  }

  render() {
    return (
      <form onSubmit={this.props.onSubmit}>
        {this.props.isLoading ? <Loader /> : this.renderForm()}
      </form>
    );
  }
}

const registerForm = props => (
  <Form onSubmit={submittedValues => props.onSubmit && props.onSubmit(submittedValues)}>
    {formApi => (
      <Register
        {...props}
        onSubmit={formApi.submitForm}
        errors={{ ...formApi.errors, ...props.errors }}
        touched={formApi.touched}
      />
    )}
  </Form>
);

registerForm.propTypes = {
  onSubmit: PropTypes.func,
  errors: PropTypes.object
};

export default registerForm;
