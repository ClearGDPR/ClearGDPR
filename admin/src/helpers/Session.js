import cookie from 'cookie';

export class Session {
  set(authResult) {
    const maxAge = 1 * 24 * 60 * 60; // 1 day long;
    document.cookie = cookie.serialize('auth', authResult, { maxAge });
    document.cookie = cookie.serialize('expires_at', authResult, { maxAge });
  }

  destroy() {
    document.cookie = cookie.serialize('auth', '', { maxAge: -1 });
    document.cookie = cookie.serialize('expires_at', '', { maxAge: -1 });
  }

  isLoggedIn() {
    let expiresAt = cookie.parse(document.cookie).expires_at || '';
    return new Date().getTime() < expiresAt;
  }
}

export default new Session();
