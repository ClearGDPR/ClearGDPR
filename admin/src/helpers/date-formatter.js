import { format } from 'date-fns';

export default date => {
  return format(new Date(date), 'DD/MM/YYYY h:mma');
};
