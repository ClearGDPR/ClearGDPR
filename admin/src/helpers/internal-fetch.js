import merge from 'lodash/object/merge';
import session from '../../helpers/Session';

const internalFetch = (url, config = {}) => {
  const internalConfig = merge(
    {
      method: 'GET',
      headers: {
        Authorization: `Bearer ${session.getToken()}`,
        'Content-Type': 'application/json'
      }
    },
    config
  );
  return fetch(url, internalConfig).then(async res => {
    if (res.status === 400 && (await res.json()).error === 'JWT token expired') {
      session.destroy();
    } else {
      return res;
    }
  });
};

export default internalFetch;
