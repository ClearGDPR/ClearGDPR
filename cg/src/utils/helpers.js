const Promise = require('bluebird');
const winston = require('winston');

const { VALID_RUN_MODES } = require('./constants');

function inControllerMode() {
  return process.env.MODE === VALID_RUN_MODES.CONTROLLER;
}

function getMyAddress() {
  return process.env.MY_ADDRESS;
}
function getTokenExpiry() {
  return process.env.JWT_TOKEN_EXPIRY || 604800;
}

const timeout = ms => new Promise(res => setTimeout(res, ms));

async function retryAsync(promise, options = {}) {
  let tries = 0;

  const maxRetries = options.maxRetries || 3;
  const interval = options.interval || 500;
  let delay = 0;

  while (tries < maxRetries) {
    await Promise.delay(delay);
    tries++;
    try {
      return await promise();
    } catch (e) {
      delay = interval * tries;
      winston.info(`Attempt no. ${tries} failed. Next attempt in ${delay}ms`);
      winston.log('debug', `Error: ${e.toString()}`);
    }
  }

  winston.error(`Operation failed after ${tries} attempts`);
}

module.exports = {
  getTokenExpiry,
  inControllerMode,
  getMyAddress,
  timeout,
  retryAsync
};
