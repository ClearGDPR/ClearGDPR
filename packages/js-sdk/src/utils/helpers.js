import { merge } from 'lodash/object';

export function parseJSON(response) {
  const contentType = response.headers.get('Content-Type');

  if (contentType.indexOf('application/json') > -1) {
    return new Promise(resolve =>
      response.json().then(json => {
        return resolve({
          status: response.status,
          ok: response.ok,
          data: json
        });
      })
    );
  } else {
    return new Promise(resolve =>
      response.text().then(text => {
        return resolve({
          status: response.status,
          ok: response.ok,
          data: text
        });
      })
    );
  }
}

export function buildRequestBody(body) {
  return body ? JSON.stringify(body) : null;
}

/**
 * Merges objects values
 * @param {Object} options objects to be merged
 * @returns {Object} New merged object based on params
 */
export function mergeOptions(...objects) {
  return merge(...objects);
}
