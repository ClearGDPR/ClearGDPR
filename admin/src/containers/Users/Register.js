import React from 'react';
import PropTypes from 'prop-types';

import Register from 'components/Users/Register';
import { PanelConsumer } from 'containers/MainLayout/PanelContext';
import { UsersConsumer } from './UsersContext';

export class RegisterContainer extends React.Component {
  static propTypes = {
    registerUser: PropTypes.func.isRequired,
    closePanel: PropTypes.func.isRequired,
    isLoading: PropTypes.bool.isRequired
  };

  state = {
    errors: {}
  };

  validatePassword(password) {
    // TODO: todo add proper validation
    if (!password) {
      return 'Field required';
    }
    if (password.length < 8) {
      return 'Password must have min. 8 characters';
    }
    return null;
  }

  onSubmit(data) {
    const { username, password } = data;

    this.props
      .registerUser(username, password)
      .then(() => this.props.closePanel())
      .catch(e =>
        this.setState({
          errors: {
            username: e.toString()
          }
        })
      );
  }

  render() {
    return (
      <Register
        errors={this.state.errors}
        validatePassword={this.validatePassword}
        onSubmit={this.onSubmit.bind(this)}
        isLoading={this.props.isLoading}
      />
    );
  }
}

export default props => (
  <PanelConsumer>
    {({ closePanel }) => (
      <UsersConsumer>
        {({ registerUser, isLoading }) => (
          <RegisterContainer
            {...props}
            isLoading={isLoading}
            registerUser={registerUser}
            closePanel={closePanel}
          />
        )}
      </UsersConsumer>
    )}
  </PanelConsumer>
);
