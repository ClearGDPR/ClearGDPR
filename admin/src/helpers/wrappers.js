import React from 'react';
import PropTypes from 'prop-types';
import lodash from 'lodash';
import { UsersProvider } from 'containers/Users/UsersContext';
import { ProcessorsProvider } from 'containers/Processors/ProcessorsContext';
import { PanelProvider } from 'containers/MainLayout/PanelContext';
import { RectificationsProvider } from 'containers/Rectifications/RectificationsContext';
import { StatsProvider } from 'containers/Dashboard/StatsContext';

export const withWrapper = Wrapper => Component => {
  let wrapper = props => (
    <Wrapper>
      <Component {...props}>{props.children}</Component>
    </Wrapper>
  );
  wrapper.propTypes = {
    children: PropTypes.node
  };
  return wrapper;
};

export const withContextProviders = lodash.flow.apply(null, [
  withWrapper(UsersProvider),
  withWrapper(ProcessorsProvider),
  withWrapper(RectificationsProvider),
  withWrapper(PanelProvider),
  withWrapper(StatsProvider)
]);
