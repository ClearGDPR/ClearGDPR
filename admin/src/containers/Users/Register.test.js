import React from 'react';
import { shallow } from 'enzyme';

import * as TestUtils from 'tests/helpers/TestUtils';

import { RegisterContainer as Register } from './Register';

const setupShallow = propOverrides => {
  const props = Object.assign(
    {
      registerUser: jest.fn().mockReturnValue(Promise.resolve()),
      closePanel: jest.fn(),
      isLoading: false
    },
    propOverrides
  );
  const component = shallow(<Register {...props} />);

  return { props, component };
};

describe('(Container) Change Password', () => {
  it('should render correctly when default props provided', async () => {
    const { component } = setupShallow();
    expect(component).toMatchSnapshot();
  });

  it('should populate errors on submit when register rejects with error', async () => {
    const { props, component } = setupShallow();

    props.registerUser.mockReturnValue(Promise.reject(new Error('Test error')));

    component.instance().onSubmit({ username: 'test.user', password: 'testPassword' });

    await TestUtils.flushPromises();
    expect(component.state()).toMatchSnapshot();
  });

  it('should close panel when registration succeeded', async () => {
    const { props, component } = setupShallow();
    component
      .instance()
      .onSubmit({ newPassword: 'testPassword', newPasswordRepeat: 'testPassword' });

    await TestUtils.flushPromises();
    expect(props.closePanel).toHaveBeenCalled();
  });

  describe('Validate password', () => {
    it('should return error when password is null or empty', async () => {
      const { component } = setupShallow();
      let result = component.instance().validatePassword('');
      expect(result).toEqual(
        expect.stringContaining('Field required')
      );
      result = component.instance().validatePassword();
      expect(result).toEqual(
        expect.stringContaining('Field required')
      );
    });

    it('should return error when password is too short', async () => {
      const { component } = setupShallow();
      let result = component.instance().validatePassword('asc');
      expect(result).toEqual(
        expect.stringContaining('Password must have min. 8 characters')
      );
      result = component.instance().validatePassword('asffas');
      expect(result).toEqual(
        expect.stringContaining('Password must have min. 8 characters')
      );
    });
  });
});
