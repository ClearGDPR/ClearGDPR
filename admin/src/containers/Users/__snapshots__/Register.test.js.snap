// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Container) Change Password should populate errors on submit when register rejects with error 1`] = `
Object {
  "errors": Object {
    "username": "Error: Test error",
  },
}
`;

exports[`(Container) Change Password should render correctly when default props provided 1`] = `
<registerForm
  errors={Object {}}
  isLoading={false}
  onSubmit={[Function]}
  validatePassword={[Function]}
/>
`;
